packages/react/test/specs/components/Menu/Menu-test.tsx

```diff
@@ -66,6 +77,19 @@ describe('Menu', () => {
       expect(menuItems.everyWhere(item => item.prop('data-foo') === 'something')).toBe(true)
     })
 
+    it('closes menu when item is clicked using spacebar', () => {
+      const menu = mountWithProvider(<Menu items={getNestedItems()} />)
+      const menuItems = menu.find('MenuItem')
+
+      menuItems.at(1).simulate('keydown', { keyCode: keyboardKey.Spacebar })
+      menuItems
+        .at(1)
+        .at(0)
+        .simulate('keydown', { keyCode: keyboardKey.Spacebar })
+
+      expect(menuItems.at(1).state('menuOpen')).toBe(false)
```

Comment:
Without the fix, the `Menu` component behaves in a weird way. The `menuOpen` state is `undefined`, instead of `true`. Upon investigation, `this.props` in `handleClick` are not what one would expect. They seem like the props of the root component. This probably causes the `menuOpen` not to be defined. I spent too much time on this, trying different ways to make the value `true` instead of `undefined`, but without luck.